config/application.rb

```diff
@@ -15,6 +15,7 @@ module Avalon
   class MissingUserId < Exception; end
 
   class Application < Rails::Application
+    require 'avalon/configuration'
```

Comment:
Letting ENV configuration merge and then testing it out with existing file configs is probably easier than excising it via cherry-pick. ENV configuration was written to be backward compatible with existing files.